src/main/java/com/varukha/onlinebookstore/service/shoppingcart/impl/ShoppingCartServiceImpl.java

```diff
@@ -0,0 +1,104 @@
+package com.varukha.onlinebookstore.service.shoppingcart.impl;
+
+import com.varukha.onlinebookstore.dto.cartitem.CartItemDto;
+import com.varukha.onlinebookstore.dto.cartitem.CreateCartItemRequestDto;
+import com.varukha.onlinebookstore.dto.cartitem.UpdateCartItemRequestDto;
+import com.varukha.onlinebookstore.dto.shoppingcart.ShoppingCartDto;
+import com.varukha.onlinebookstore.mapper.CartItemMapper;
+import com.varukha.onlinebookstore.mapper.ShoppingCartMapper;
+import com.varukha.onlinebookstore.model.CartItem;
+import com.varukha.onlinebookstore.model.ShoppingCart;
+import com.varukha.onlinebookstore.model.User;
+import com.varukha.onlinebookstore.repository.book.BookRepository;
+import com.varukha.onlinebookstore.repository.cartitem.CartItemRepository;
+import com.varukha.onlinebookstore.repository.shoppingcart.ShoppingCartRepository;
+import com.varukha.onlinebookstore.service.shoppingcart.ShoppingCartService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+@RequiredArgsConstructor
+public class ShoppingCartServiceImpl implements ShoppingCartService {
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final ShoppingCartMapper shoppingCartMapper;
+    private final CartItemMapper cartItemMapper;
+    private final CartItemRepository cartItemRepository;
+    private final BookRepository bookRepository;
+
+    @Override
+    public void createShoppingCart(User user) {
+        ShoppingCart shoppingCart = new ShoppingCart();
+        shoppingCart.setUser(user);
+        shoppingCartRepository.save(shoppingCart);
+    }
+
+    @Transactional
+    @Override
+    public CartItemDto createCartItem(CreateCartItemRequestDto requestDto) {
+        CartItem existedCartItem = getCartItemByBookId(requestDto.getBookId());
+        if (existedCartItem == null || !existedCartItem.getBook().getId()
```

Comment:
how is it possible that book id is other than `requestDto.getBookId()` if you are searching by this id?